web-app/HeimScripts/heatmap/summary.R

```diff
@@ -29,14 +29,43 @@
 library(jsonlite)
 library(gplots)
 
-main <- function(phase)
+main <- function(phase = NULL)
 {
+  check_input(loaded_variables, phase)
   data_measurements <- extract_measurements(loaded_variables)
-  produce_summary_stats(data_measurements, phase)
+  summary_stats_json <- produce_summary_stats(data_measurements, phase)
+  write_summary_stats(summary_stats_json)
   produce_boxplot(data_measurements, phase)
   return(list(summary_stats = "Finished")) #right now a non-empty list is expected as a return.
```

Comment:
Current agreement with the Frontend is that we pass "messages" list. A vector of Strings. These messages will be displayed on the frontend and they also serve as nice filler for the list we have to return from main function anyway.